src/ucs/sys/sock.c

```diff
@@ -179,6 +185,47 @@ int ucs_socket_max_conn()
     }
 }
 
+int ucs_socket_max_iov()
+{
+    static int max_iov = -1;
+
+#if defined(IOV_MAX)
+    max_iov = IOV_MAX;
+#elif defined(_SC_IOV_MAX)
+    if (max_iov == -1) {
+        max_iov = sysconf(_SC_IOV_MAX);
+        if (max_iov == -1) {
+            max_iov = 1;
+        }
+    }
+#elif defined(UIO_MAXIOV)
+    max_iov = UIO_MAXIOV;
+#else
+    max_iov = 1024;
```

Comment:
documented it

the value is chosen because latest Linux kernels, freebsd's value reported from sysconf(_SC_IOV_MAX), glibc define it as 1024:
https://elixir.bootlin.com/linux/v5.2.1/source/include/uapi/linux/uio.h#L28
https://elixir.bootlin.com/musl/v1.1.23/source/include/limits.h